File: archive/p/perl/factorial.pl
#!/usr/bin/env perl
use v5.42;

sub usage {
    say "Usage: please input a non-negative integer";
    exit;
}

my ($number) = @ARGV;
usage() unless defined $number && $number =~ /\A\d+\z/;

my $factorial = 1;
$factorial *= $_ for 2 .. $number;

say $factorial;

File: archive/p/php/factorial.php
<?php

$input = null;
if (
    sizeof($argv) < 2 ||
    !is_numeric($argv[1]) ||
    intval($argv[1]) < 0
) {
    echo ("Usage: please input a non-negative integer\n");
    exit(0);
}

$factorial = 1;
$input = intval($argv[1]);

for ($x = $input; $x >= 1; $x--) {
    $factorial = $factorial * $x;
}

echo ("$factorial\n");
